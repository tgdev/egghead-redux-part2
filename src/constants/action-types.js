export const ACTION_ADD_TODO = 'ADD_TODO';
export const ACTION_TOGGLE_TODO = 'TOGGLE_TODO';

export const ACTION_SET_VISIBILITY_FILTER = 'SET_VISIBILITY_FILTER';
