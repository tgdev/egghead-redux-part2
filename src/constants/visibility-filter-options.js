export const FILTER_OPTION_ALL = 'SHOW_ALL';
export const FILTER_OPTION_ACTIVE = 'SHOW_ACTIVE';
export const FILTER_OPTION_COMPLETED = 'SHOW_COMPLETED';
