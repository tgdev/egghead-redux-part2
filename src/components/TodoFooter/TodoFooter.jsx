import React from 'react';

import { FILTER_OPTION_ALL, FILTER_OPTION_ACTIVE, FILTER_OPTION_COMPLETED } from 'constants/visibility-filter-options';

import FilterLinkContainer from 'components/FilterLink/FilterLinkContainer';

const TodoFooter = () => {
  return (
    <p>Show:
      {' '}
      <FilterLinkContainer
        filter={FILTER_OPTION_ALL}>
        All
      </FilterLinkContainer>
      {' '}
      <FilterLinkContainer
        filter={FILTER_OPTION_ACTIVE}>
        Active
      </FilterLinkContainer>
      {' '}
      <FilterLinkContainer
        filter={FILTER_OPTION_COMPLETED}>
        Completed
      </FilterLinkContainer>
    </p>
  );
};

export default TodoFooter;
