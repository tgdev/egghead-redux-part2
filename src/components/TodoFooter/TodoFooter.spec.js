import React from 'react';
import { shallow } from 'enzyme';

import { FILTER_OPTION_ALL } from 'constants/visibility-filter-options';

import TodoFooter from './TodoFooter';

describe('TodoFooter component', () => {
  const mockClickHandler = jest.fn();
  const mockVisibilityFilter = FILTER_OPTION_ALL;

  afterEach(() => {
    mockClickHandler.mockReset();
  });

  it('renders TodoFooter correctly', () => {
    const wrapper = shallow(<TodoFooter visibilityFilter={mockVisibilityFilter} handleFilters={mockClickHandler} />);
    expect(wrapper).toMatchSnapshot();
  });

});
