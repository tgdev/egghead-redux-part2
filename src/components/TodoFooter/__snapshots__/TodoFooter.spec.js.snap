// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFooter component renders TodoFooter correctly 1`] = `
<p>
  Show:
   
  <Connect(FilterLink)
    filter="SHOW_ALL"
  >
    All
  </Connect(FilterLink)>
   
  <Connect(FilterLink)
    filter="SHOW_ACTIVE"
  >
    Active
  </Connect(FilterLink)>
   
  <Connect(FilterLink)
    filter="SHOW_COMPLETED"
  >
    Completed
  </Connect(FilterLink)>
</p>
`;
